src/dmd/expressionsem.d

```diff
@@ -9022,8 +9022,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             if (checkNewEscape(sc, exp.e2, false))
                 return setError();
 
-            exp.op = TOK.concatenateElemAssign;
-            exp.e2 = exp.e2.castTo(sc, tb1next);
+            exp = new CatElemAssignExp(exp.loc, exp.type, exp.e1, exp.e2.castTo(sc, tb1next));
```

Comment:
We're having problems with DMD using way too much memory.  This will likely add to the problem, but I don't see a better way if we want `op` to be const, which we do.

Reply:
It's not an often used operator.